hypothesis/via
# PDF.js update 2/n - Add tools to update PDF.js to the latest version

This PR adds a `tools/update-pdfjs` script which fetches the latest version of the PDF.js viewer from the official PDF.js repo on GitHub and generates the Jinja template for the PDF.js viewer based on viewer.html. It also adds updated logic in `static/scripts/pdfjs-init.js` for loading the PDF specified in the URL that works with PDF.js v2.

Aside from updating to the current PDF.js version, this is intended to make it much easier to update to new PDF.js releases in future in several ways:

- By adding scripts to automate the process
- By using only "public" APIs in the PDF.js init script instead of the monkey-patching that `static/pdfjs/proxy.js` currently does
- By having a clearer separation between vendor files, generated code and manually written code

If we end up replacing Via entirely, these scripts should still be useful as we can copy them to whatever repo the replacement ends up living in. We might also consider creating an npm package out of the viewer that we can use in the browser extension, Via. and other repos.

**To test this out:**

1. Combine this branch with https://github.com/hypothesis/via/pull/178 (that is, check out this branch, then do `get merge <branch for #178>`)
2. Run `tools/update-pdfjs`
3. Restart Via
4. Go to a PDF in Via with the "pdfjs2" feature flag enabled (eg. http://localhost:9080/https://s3.amazonaws.com/happiness-report/2019/WHR19.pdf?via.features=pdfjs2). I suggest opening the Network tab in your browser dev tools and checking the option to disable the cache before doing this.
5. Check the console logs and you should see that a message is printed indicating that PDF.js v2.x is being used.
